### DISCUSSION TITLE
Is there any concept of having a prestashop security download channel, to download ONLY security patches?

### DISCUSSION THREAD

**Discussion by ProteanCode** [PrestaShop/PrestaShop#31756] [Q&A] [answered]
I used to see that upgrading prestashop is real pain especially for templates.

Is there a way to download only security fixes? In 99% of cases I don't need any other improvements / new functionalities. I am just looking to keep data secured and want my head free from fixing anything a typical upgrade breaks

**Comment by matks** [comment]
Hello @ProteanCode you can do it but it might be tedious: you need to look at the list of security issues here https://github.com/PrestaShop/PrestaShop/security/advisories

and then look inside the Changelog of each version to find the exact Pull Request that fix the problem

For example for 8.0.1 you look at https://github.com/PrestaShop/PrestaShop/releases/tag/8.0.1 and see `https://github.com/advisories/GHSA-9qgp-9wwc-v29r: Potential Information exposure in the upload directory (by @atomiix)` then you look at this history https://github.com/PrestaShop/PrestaShop/compare/8.0.0...8.0.1 and find the PR

I dont know if there is a more efficient way

**Reply by ProteanCode** [reply]
What if we could create a separate branch straight for security fixes that would be merged into the core separately?

The only problems with prestashop that I had over last few years are upgrade related problems; Every hacked customer gets hacked partially because when they ask how much will it cost to upgrade and hear the estimated working hours, then suddenly they no longer care about any security measurements. Since there is no budget for that kind of work, bad situations happens and I always warn my customers about it. 

Something like security channel fixes would be a truly LTS-like solution.

Heck I would probably even pay like 10$ monthly for fully automated plugin that would merge these changes into a core files. It sounds like an automated enterprise support.

**Reply by matks** [reply]
> Something like security channel fixes would be a truly LTS-like solution.

As you have guessed at the end of your message, the problem is the cost.

We have to provide bugfixes in regular patches like 8.0.2 (that was released yesterday) so this "security only" branch would be extra work for the project.

For example today we maintain 4 branches for security: 1.7.8.x, 8.0.x, 8.1.x and develop. If tomorrow a critical vulnerability is reported we will need to build a fix, then adapt it for the 4 branches and then perform 4 releases. Performing 4 releases requires developer time and QA time, and if during the release test a blocker bug is found, then we need to fix it then re-start the release test again... so one more branch would be 20% more work for us.

> Heck I would probably even pay like 10$ monthly for fully automated plugin that would merge these changes into a core files. It sounds like an automated enterprise support.

I think the only way to make it happen and it be cost-effective is indeed a paid solution 😄 but then that is outside of my scope: if a company sees this message, they can seize this opportunity but I personally have no power to start a new paid service for PrestaShop

**Reply by tenenick** [reply]
Totally agree with ProteanCode. I am sure there are many users out there who think the same.
But unfortunately there is no easy solution. I guess when Prestahop started security issues weren't that important as they are now. Not important enough to have a security branch from the beginning, which would be the real solution.
Nice day to everyone and many thanks to the outstanding work you are doing to make Prestahop possible !

**Comment by matks** [comment, marked as answer]
Hello guys, good news for you : we now have changed how we deliver security fixes. When there is a major security issue; we now deliver a new version which contains only these changes.

I know it's not 100% what was requested but it is an improvement 😉 

That new idea was applied for PrestaShop 8.1.4 and PrestaShop 8.1.6 and I just submitted a PR to update the process to describe it https://github.com/PrestaShop/open-source/pull/261

I close this discussion then

**Reply by tenenick** [reply]
thanks a lot Mathieu for me a great improvement.
Have a nice day
